google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/sdk/CloudSdkPanel.java

```diff
@@ -347,7 +359,19 @@ protected void textChanged(DocumentEvent event) {
   private void setManagedSdkUiAvailable(boolean available) {
     if (ServiceManager.getService(PluginInfoService.class).shouldEnable(GctFeature.MANAGED_SDK)) {
       enableAutomaticUpdatesCheckbox.setEnabled(available);
-      checkForUpdatesHyperlink.setVisible(available);
+      // only make it visible if managed SDK is active, not currently installing or updating, and
+      // not
+      // up-to-date.
+      CloudSdkService cloudSdkService = CloudSdkService.getInstance();
+      if (cloudSdkService instanceof ManagedCloudSdkService
```

Comment:
if we need to check this maybe we should push it into the CloudSdkService interface?

Reply:
That's a good discussion question, we've been thinking about it a bit and so far decided CloudSdkService shouldn't grow to include update() and isUpToDate() methods since they are meaningless for custom SDK and we don't foresee any future uses at all. Extending to another interface (like UpdateableCloudSdkService) would still have the need for casting. Only places where we do that are those where we are explicitly strongly coupled to ManagedCloudSdkService and won't expect another service (like here in settings.)

Reply:
I was actually thinking something way less thoughtful. :)

e.g. `CloudSdkService.isManaged()`

Reply:
But in general you are definitely correct that simple "path" based custom SDK and managed SDK are two different beasts hence couple of places where only can work with managed one. Since the changes introduced a lot of refactoring, we tried to keep existing functionality based on previous interface, but once we're done with managed SDK and things go well we may try to split interface into two parts for further trasparency.